controllers/controllers/workloads/cfapp_controller_test.go

```diff
@@ -240,6 +249,29 @@ var _ = Describe("CFAppReconciler", func() {
 					Expect(reconcileErr).To(MatchError(failsOnPurposeErrorMessage))
 				})
 			})
+
+			When("fetch vcap services Secret returns an error", func() {
+				BeforeEach(func() {
+					secretErr = errors.New(failsOnPurposeErrorMessage)
+					_, reconcileErr = cfAppReconciler.Reconcile(ctx, req)
+				})
+
+				It("should return an error", func() {
+					Expect(reconcileErr).To(MatchError(failsOnPurposeErrorMessage))
+				})
+			})
+
+			When("fetch vcap services Secret returns a NotFoundError and create Secret returns an error", func() {
```

Comment:
I would suggest rephrasing this context to something like

```
When("vcap services secret does not exist"
   It("creates the vcap services secret"
  When("cresting the vcap services secret fails
    It("returns an error
```
```

Reply:
Isn't "creates the vcap services secret" part of the happy path though? This whole block is inside a When block for "on the unhappy path".